packages/core/src/types.ts

```diff
@@ -1893,6 +1893,7 @@ export interface ActorSystem<T extends ActorSystemInfo> {
   _unregister: (actorRef: AnyActorRef) => void;
   _set: <K extends keyof T['actors']>(key: K, actorRef: T['actors'][K]) => void;
   get: <K extends keyof T['actors']>(key: K) => T['actors'][K] | undefined;
+  root?: T['actors'][keyof T['actors']];
```

Comment:
Is there a way to restrict this to only be the type of the root actor?

Reply:
Not yet (to my knowledge); best to have this as required and `AnyActorRef` for now

Reply:
Looks like there are a couple options here:

1. add a `root` property + generic param for it to `ActorSystemInfo` so that we can reference it directly on `T`
2. add a new generic param to `ActorSystem` that `extends AnyActorRef` so we can specify the root type without adding a generic param to `ActorSystem`. would also need to add a generic param to `Interpreter` so it can be passed in to `ActorSystem`

Reply:
@with-heart Thank you for these other suggestions! For simplicity, I made the type required and `AnyActorRef` but I'd be interested in taking more time to explore some alternatives as I dig deeper into the code and types.

Reply:
@with-heart has good ideas about what it would take to support this at type-level here. And with those changes in mind - I'm not quite convinced that this is a feature that we'd like to land.

To fully support this we **need** type-level support. Types nowadays are not a nice-to-have, they are the baseline. 

So with that in mind, I'm not sure if this will make the overall usage that much simpler once we demand from people to type this. At the definition site you will have to remember how to type this - is this a second param? first? something else? Treating this just like every other registerable actor is pretty neat because the user doesn't have to learn any new concept, nor API.

I see one advantage of this though: this particular actor wouldn't have to be nullable because it's guaranteed to exist. 

It would be nice if the PR's description would motivate this addition beyond not having to use the explicit ID at the call site. I feel like this alone is really a tiny improvement. 

Either way, I would like to get this on hold. I think that we should first land improved types for the system in its current shape. Only with that base in place, we'll actually be able to tell what's the cost of this feature and if it's worth it. The change itself is purely additive so it can land at just any time in the future.

Reply:
@Andarist Thanks for thinking this one through a bit more and thanks again @with-heart again for your type suggestions. I'll hold off on working on this until we find a way to type the root actor. In the meantime, I'll add mention in the Docs about how to set and obtain the root actor's `systemId` when sending events.